src/coreclr/src/ilasm/grammar_after.cpp

```diff
@@ -1542,18 +1534,17 @@ char* AsmParse::fillBuff(__in_opt __nullterminated char* pos)
     }
     else
     {
-        if(((penv->curPos[0]&0xFF)==0xEF)&&((penv->curPos[1]&0xFF)==0xBB)&&((penv->curPos[2]&0xFF)==0xBF))
+        if((penv->curPos[0] & 0xFF) == 0xEF && (penv->curPos[1] & 0xFF) == 0xBB && (penv->curPos[2] & 0xFF) == 0xBF)
```

Comment:
It may be nice to delete `IsItUnicode` and `IsTextUnicode` in the PAL, and just inline the check for UTF16 BOM from `IsTextUnicode` above so that it looks similar to this check for UTF8 BOM. It would fix https://github.com/dotnet/runtime/issues/4778.

Reply:
There is a corner case which is not handled by `IsTextUnicode` for UTF-16 LE, i.e. after FF and FE, position 2 and position 3 should not be 0. In terms of inlined version:

```diff
- if(iPutToBuffer >= 2 && (penv->curPos[0] & 0xFF) == 0xFF && (penv->curPos[1] & 0xFF) == 0xFE)
+ if(iPutToBuffer >= 4 && (penv->curPos[0] & 0xFF) == 0xFF && (penv->curPos[1] & 0xFF) != 0xFE && (penv->curPos[2] & 0xFF) != 0x00 && (penv->curPos[3] & 0xFF) != 0x00)
```

because:

```sh
# 0xFFFE followed by 0x00 and 0x00 is UTF-32
$ printf "\xff\xfe\x00\x00" > /tmp/x && file /tmp/x
/tmp/x: Unicode text, UTF-32, little-endian

# any non-zero value after 0xFFFE is UTF-16 LE
$ printf "\xff\xfe\x00\x01" > /tmp/x && file /tmp/x
/tmp/x: Little-endian UTF-16 Unicode text, with no line terminators
```